core/templates/vector.h

```diff
@@ -145,25 +146,31 @@ class Vector {
 		return ret;
 	}
 
-	Vector<T> slice(int p_begin, int p_end) const {
+	Vector<T> slice(int p_begin, int p_end = INT_MAX) const {
 		Vector<T> result;
 
-		if (p_end < 0) {
-			p_end += size() + 1;
-		}
+		const int s = size();
 
-		ERR_FAIL_INDEX_V(p_begin, size(), result);
-		ERR_FAIL_INDEX_V(p_end, size() + 1, result);
+		int begin = CLAMP(p_begin, -s, s);
+		if (begin < 0) {
+			begin += s;
+		}
+		int end = CLAMP(p_end, -s, s);
+		if (end < 0) {
+			end += s;
+		}
 
-		ERR_FAIL_COND_V(p_begin > p_end, result);
+		if (begin > end) {
+			SWAP(begin, end);
+		}
```

Comment:
I think this assumes that the user put something wrong as the argument.

Python & JavaScript both return an empty array in this case. That makes more sense as there _is_ nothing between the two indices.